### DISCUSSION TITLE
Can't play  Ogg/Opus files

### DISCUSSION THREAD

**Discussion by ushistiy** [jonsafari/mocp#39] [General]
(5225!) not a vorbis file
(481961!) not a vorbis file 
(510031!) not a vorbis file 

Moc does NOT play opus files encasulated in ogg.

```
#  mocp -V

          This is : Music On Console
          Version : 2.6-alpha3
         Revision : 2933
            Built : Aug 10 2023 19:40:33
    Compiled with : ALSA Network streams resample
       Running on : Linux 6.1.22-gentoo-dist x86_64
           Author : Damian Pietras
         Homepage : http://moc.daper.net/
           E-Mail : mocmaint@daper.net
        Copyright : (C) 2003-2016 Damian Pietras and others
          License : GNU General Public License, version 2 or later

```

**Comment by tomaszg7** [comment]
MOC supports Opus via ffmpeg plugin. Is it installed? Anyway, you should report it on MOC official forum, not some random fork. If you feel adventurous you can try my experimental code which can be found https://gitlab.com/tomaszg/mocp It supports Opus natively via libopusfile.

**Reply by ushistiy** [reply]
Normal opus files can be played normally. But if they are encapsulated in ogg, then the program considers that these are vorbis files. ffmeg is most likely installed, as it was rebuilt. Thanks, I'll try your fork.

**Reply by ushistiy** [reply]
Fork doesn't work either.

`./configure --enable-debug=no --with-oss=no`

```
mocp -V

          This is : Music On Console
          Version : 2.6-alpha3
            Built : Aug 11 2023 02:26:53
    Compiled with : ALSA Network streams resample MPRIS/D-Bus
       Running on : Linux 6.1.22-gentoo-dist x86_64
           Author : Damian Pietras
         Homepage : http://moc.daper.net/
           E-Mail : mocmaint@daper.net
        Copyright : (C) 2003-2016 Damian Pietras and others
          License : GNU General Public License, version 2 or later
```

**Comment by tomaszg7** [comment]
Isn't it just an issue of an extension then? Opus files are almost always encapsulated in ogg, but they have .opus extension. You could try using PreferredDecoders in MOC, pointing all ogg files to ffmpeg, or using mime. I didn't try that. Or just rename them to .opus ;)

**Reply by ushistiy** [reply]
I can't just rename the file extension, the program must determine the file by mime types. If you want to reproduce the problem, here is the command to get such a file.

yt-dlp -x --embed-metadata --remux-video "ogg" https://any-youtube-link

**Reply by tomaszg7** [reply]
Like I said, it is an issue of `PreferedDecoder`. Since I have made some changes to it in my config file the ffmpeg decoder kicks in and plays it just fine. 
```
Aug 11 19:34:36.744840: ffmpeg.c:746 ffmpeg_open_internal(): FFmpeg thinks 'Tankowanie Noca [Lv7El77HF1g].ogg' is format(codec) 'ogg(opus)'
```
So, copy the default values of PreferedDecoder from config.example and change ogg to be handled by ffmpeg by default. Or if you feel adventurous you may just set 
```
PreferredDecoders=ogg(vorbis)
```
It will override all default values, but that might be even better as the defaults are quite conservative and not always best for the modern case.
 
However the file I got in this way *is* a normal opus file. I believe you really can safely rename them all to .opus. Ogg is the native cointainer for opus streams, the extension opus is just for convenience. With this extension either my Opus decoder or ffmpeg plays it nicely.

**Reply by ushistiy** [reply]
I'll try it. Thank you very much.

**Reply by ushistiy** [reply]
I added the line to the file /home/user/.moc/config

`PreferredDecoders = ogg(ffmpeg)`

I can confirm that everything works.  Even streaming broadcast ogg/opus in Isecast.
